### DISCUSSION TITLE
I want a code sample to test s3 upload using role, not access key.

### DISCUSSION THREAD

**Discussion by sangbinlee** [thuliteio/doks#1057] [Q&A]
Since the application runs in the eks environment, we have been guided to use a role when authenticating.
I need s3 file upload sample code using role.

When uploading file to s3
Use a role, not an access key.

The code below is the code used when testing a successful file upload to the S3 bucket using the ACCESS KEY.

		// Upload a local file to Amazon S3
			S3AsyncClient s3Client = S3AsyncClient.crtCreate();
			PutObjectResponse putObjectResponse =
			      s3Client.putObject(req -> req.bucket(BUCKET_NAME)
			                                   .key(downloadFileName),
			                        AsyncRequestBody.fromFile(Paths.get(FILE_PATH)))
			              .join();

**Comment by h-enk** [comment]
Raise an issue /get help at [bep/s3deploy](https://github.com/bep/s3deploy)?
